python/cudf/cudf/core/multiindex.py

```diff
@@ -1539,13 +1418,9 @@ def memory_usage(self, deep=False):
         return n
 
     def difference(self, other, sort=None):
-        temp_self = self
-        temp_other = other
-        if hasattr(self, "to_pandas"):
-            temp_self = self.to_pandas()
         if hasattr(other, "to_pandas"):
-            temp_other = self.to_pandas()
-        return temp_self.difference(temp_other, sort)
+            other = other.to_pandas()
+        return self.to_pandas().difference(other, sort)
```

Comment:
I see this function is never tested, or used. Any knowledge of why it is here?

Reply:
`Index.difference` exists in pandas, so this method probably should exist. The fact that tests are missing is probably why this clearly buggy implementation wasn't caught before, I can add some tests in this PR.